lib/src/platform/src/push_notification_events_native.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:async';
+import 'dart:io' as io show Platform;
+
+import 'package:flutter/services.dart';
+
+import '../../generated/platform_constants.dart';
+import '../../push_notifications/src/push_notification_events.dart';
+import '../../push_notifications/src/remote_message.dart';
+import '../platform.dart';
+import 'background_android_isolate_platform.dart';
+
+class PushNotificationEventsNative implements PushNotificationEvents {
+  Future<bool> Function(RemoteMessage message)?
+      onShowNotificationInForegroundHandler;
+  StreamController<RemoteMessage> onMessageStreamController =
+      StreamController();
+
+  /// Controller used
+  StreamController<RemoteMessage> onNotificationTapStreamController =
+      StreamController();
+
+  @override
+  Future<RemoteMessage?> get notificationTapLaunchedAppFromTerminated =>
+      Platform.methodChannel.invokeMethod(
+          PlatformMethod.pushNotificationTapLaunchedAppFromTerminated);
+
+  @override
+  Stream<RemoteMessage> get onMessage => onMessageStreamController.stream;
+
+  @override
+  Stream<RemoteMessage> get onNotificationTap =>
+      onNotificationTapStreamController.stream;
+
+  @override
+  void setOnShowNotificationInForeground(
+      Future<bool> Function(RemoteMessage message) callback) {
+    onShowNotificationInForegroundHandler = callback;
+  }
+
+  /// An internal method that is called from the Platform side to check if the user
+  /// wants notifications to be shown when the app is in the foreground.
+  Future<bool> showNotificationInForeground(RemoteMessage message) async {
+    if (onShowNotificationInForegroundHandler == null) {
+      return false;
+    }
+    return onShowNotificationInForegroundHandler!(message);
+  }
+
+  BackgroundMessageHandler? _onBackgroundMessage;
+
+  /// Implementation of setOnBackgroundMessage. For more documentation,
+  /// see [PushNotificationEvents.setOnBackgroundMessage]
+  void setOnBackgroundMessage(BackgroundMessageHandler handler) async {
+    _onBackgroundMessage = handler;
+    if (io.Platform.isAndroid) {
+      try {
+        await BackgroundIsolateAndroidPlatform.methodChannel
+            .invokeMethod(PlatformMethod.pushSetOnBackgroundMessage);
+      } on MissingPluginException {
```

Comment:
The android side is currently only listening for this method when the isolate was launched **manually** by the Android side. So this code throwing an exception is actually the norm when an app starts, but happens once per launch. The only time this method channel works is when the Activity is not running. So currently, it is intentional that user does not see this error when the app launches normally. Unfortunately, if something breaks in the case where the app is not launched normally (isolate launched manually), we won't know about it. I initially opted to not added extra logic code to handle this on the platform side, since it can be kept entirely in the dart side.

An easy solution to this is to add a listener on the android side to listen for this when the isolate was launched with the Activity (a normal app/ **not manually**), and do nothing as a result of this method call, but this would add complexity/ code. To avoid swallowing errors, I'll implement this.

Reply:
I've made the above changes in: 70251990f2c6e96cecf11bbe74096cf84929c712, "Replace exception catching with background method call handler which ignores the method call".

Reply:
Thanks. I can see that required a fair bit of engineering to work around, but hopefully worth it. 🤞